syntax = "proto3";

package ledcontrol;

service LedControl {
  rpc SetLedState (LedStateRequest) returns (LedStateResponse) {}
  rpc GetLedState (GetLedStateRequest) returns (LedStateResponse) {}
}

message LedStateRequest {
  string pattern = 1;  // Pattern name or description
  map<string, string> parameters = 2;  // Additional parameters for the pattern
}

message GetLedStateRequest {
  // Empty request to get current state
}

message LedStateResponse {
  bool success = 1;
  string message = 2;
  string current_pattern = 3;
}